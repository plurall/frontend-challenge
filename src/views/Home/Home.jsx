import React, {Component} from 'react'

import { Link } from 'react-router-dom';
import { SomosClient } from 'utils'
import logoURL from '../../assets/logo-spotify.png'

export default class Home extends Component {
  state = {}

  client = new SomosClient()

  render() {
    return (
      <>
        {/* <SubHeader
          breadcrumb={[{ text: 'Home' }]}
          heading="Somos Front-end Challange"
        /> */}
        <div className="home">
          <Link to="/busca">
            <img src={logoURL} alt="Spotify" className="logo-spotify" />
          </Link>
        </div>
      </>
    )
  }
}

