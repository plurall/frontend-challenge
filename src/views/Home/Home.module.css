.buttons {
  margin-top: 50px;
  display: flex;
  justify-content: center;
}

.button {
  color: #fff;
  background: var(--color-green);
  border-width: 0;
  border-style: hidden;
  border-radius: 20px;
  padding: 10px 30px;
  width: max-content;
  font-size: 14px;
  font-weight: 700;
  text-transform: uppercase;
  outline: 0;
  text-decoration: none;
  transition: all 200ms ease;
}

.button:hover {
  background-color: var(--color-dark-green);
}
