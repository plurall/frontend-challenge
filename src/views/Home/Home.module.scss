@import 'shared.module.scss';

.wrapper {
  max-width: $max-page-content-width;
  margin: 0 auto;
  padding: 2rem 1rem;
  font-family: 'Roboto', sans-serif;
  min-height: calc(100vh - var(--height-nav-bar) - var(--height-footer));

  display: flex;

  @media (max-width: $mobile-breakingpoint) {
    padding: 3rem 0.5rem;
  }
}

.content {
  margin: auto;
  height: 25rem;
  width: 100%;
  max-width: 55rem;

  display: grid;
  grid-template-columns: 1fr 18rem;
  grid-template-rows: 1fr 1fr;
  grid-template-areas:
    'image title'
    'image button';
  grid-gap: 1rem 4rem;

  @media (max-width: $mobile-breakingpoint) {
    height: initial;
    max-width: 35rem;

    display: flex;
    flex-direction: column;
    grid-gap: initial;
  }
}

.contentImage {
  grid-area: image;
  width: 100%;
  height: 100%;
  background-image: var(--image);
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;

  @media (max-width: $mobile-breakingpoint) {
    aspect-ratio: 1.5/1;
  }
}

.title {
  grid-area: title;
  font-size: 1.75rem;
  font-weight: 600;
  color: var(--color-on-background-highlight);
  align-self: flex-end;

  @media (max-width: $mobile-breakingpoint) {
    margin: 2rem 0;
    text-align: center;
  }
}

.brandImage {
  display: inline-block;
  height: 2.5rem;
  aspect-ratio: 3.33/1;
  background-image: var(--image);
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;

  transform: translateY(25%);
}

.button {
  grid-area: button;
  height: 3rem;
  width: 100%;
  align-self: center;
  padding: 0 1rem;
  font-size: 1rem;
  font-weight: 500;
  text-decoration: none;
  border-radius: 0.5rem;
  background-color: var(--color-primary);
  color: var(--color-on-primary);
  cursor: pointer;
  transition-duration: 0.2s;

  display: flex;
  align-items: center;
  justify-content: center;

  &:hover {
    filter: brightness(1.15);
  }
}
