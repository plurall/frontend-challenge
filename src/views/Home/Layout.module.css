.nav-bar {
  background-color: var(--color-dark);
  color: var(--color-white);
  font-size: 30px;
  font-weight: bolder;
  height: 400px;
  padding: 30px;
  margin-bottom: 60px;
  border-radius: 0 0 150px 0;
  display: flex;
  flex-direction: column;
  margin-top: 100px;
}

.profile-box {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.profile-picture {
  border-radius: 50%;
  width: 150px;
  height: 150px;
  margin: 10px;
}

.user-email {
  font-size: 20px;
  font-weight: 100;
  padding-bottom: 20px;
}

.user-box-btn {
  padding: 10px 20px;
  background-color: var(--color-green);
  border: none;
  border-radius: 10px;
  font-size: 20px;
  letter-spacing: 1px;
  font-weight: bolder;
  color: var(--color-white);
  transition: 0.2s;
  cursor: pointer;
  position: relative;
}

.user-box-btn:hover {
  background-color: var(--color-red);
  border-radius: 20px;
}

.loading-element {
  border-top: 5px solid var(--color-white);
  border-radius: 50%;
  width: 100px;
  height: 100px;
  animation: spinner 1s infinite linear;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(0%, 0%);
}

.transition-in {
  animation: fade linear 0.3s;
}

@keyframes fade {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes spinner {
  from {
    transform: rotate(0);
  }

  to {
    transform: rotate(180);
  }
}
