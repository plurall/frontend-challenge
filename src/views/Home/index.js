/* eslint-disable linebreak-style */
export { default } from './Home'
