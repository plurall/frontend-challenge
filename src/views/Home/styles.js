import styled from 'styled-components';

export const StyleButtonStart = styled.button`
  padding: 20px 40px;
  border-radius: 38px;
  font-size: 1.5rem;
  background: #655aa3;
  border: none;
  color: white;
  box-shadow: 1px 1px 14px #777777;
  font-weight: 600;
  cursor: pointer;
  transition: background-color 150ms ease-in;

  :hover {
    background: #39335a;
  }
`;