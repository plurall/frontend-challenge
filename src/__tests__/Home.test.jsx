import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Home from '../views/Home'


it('Home renders correctly', () => {
  const tree = renderer
    .create(<Router>
		  	<Home/>
			</Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
