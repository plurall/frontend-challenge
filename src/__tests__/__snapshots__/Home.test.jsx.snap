// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="home"
>
  <a
    href="/busca"
    onClick={[Function]}
  >
    <img
      alt="Spotify"
      className="logo-spotify"
      src="logo-spotify.png"
    />
  </a>
</div>
`;
